notification/src/config/config.js

```diff
@@ -1,81 +1,109 @@
+    secretHmacKey: adyenConfig.secretHmacKey,
+    enableHmacSignature: adyenConfig.enableHmacSignature !== 'false',
   }
 }
 
-function getFileConfig() {
-  let fileConfig = {}
-  try {
-    fileConfig = require(configPath) // eslint-disable-line
-  } catch (err) {
-    // config file was not provided
-  }
+function getAllCtpProjectKeys() {
+  return Object.keys(config.commercetools)
+}
 
-  return fileConfig
+function getAllAdyenMerchantAccounts() {
+  return Object.keys(config.adyen)
 }
 
-module.exports = function load() {
-  /**
-   * Load configuration from several sources in this order (last has highest priority):
-   * - default config
-   * - file config
-   * - ctp credentials from env variables
-   * - ctp config
-   * - env config
-   */
-  if (config === undefined) {
-    config = merge(
-      getEnvConfig(),
-      { ctp: getCTPEnvCredentials() },
-      { adyen: getAdyenCredentials() },
-      getFileConfig()
+function loadAndValidateConfig() {
+  try {
+    config = JSON.parse(process.env.ADYEN_INTEGRATION_CONFIG)
+  } catch (e) {
+    throw new Error(
+      'Adyen integration configuration is not provided in the JSON format'
+    )
+  }
+  const numberOfCtpConfigs = Object.keys(config.commercetools).length
+  const numberOfAdyenConfigs = Object.keys(config.adyen).length
+  if (numberOfCtpConfigs === 0)
+    throw new Error(
+      'Please add at least one commercetools project to the config'
     )
+  if (numberOfAdyenConfigs === 0)
+    throw new Error(
+      'Please add at least one Adyen merchant account to the config'
+    )
+
+  for (const [ctpProjectKey, ctpConfig] of Object.entries(
+    config.commercetools
+  )) {
+    if (!ctpConfig.clientId || !ctpConfig.clientSecret)
+      throw new Error(
+        `[${ctpProjectKey}]: CTP project credentials are missing. ` +
+          'Please verify that all projects have projectKey, clientId and clientSecret'
+      )
+  }
 
+  for (const [adyenMerchantAccount, adyenConfig] of Object.entries(
+    config.adyen
+  )) {
     if (
-      !config.ctp.projectKey ||
-      !config.ctp.clientId ||
-      !config.ctp.clientSecret
+      adyenConfig.enableHmacSignature !== 'false' &&
+      isEmpty(adyenConfig.secretHmacKey)
     )
-      throw new Error('CTP project credentials are missing')
-
-    if (config.adyen.enableHmacSignature && isEmpty(config.adyen.secretHmacKey))
       throw new Error(
-        'The "ADYEN_SECRET_HMAC_KEY" environment variable is missing to be able to verify notifications, ' +
-          'please generate a secret HMAC key in Adyen Customer Area ' +
-          'or set "ADYEN_ENABLE_HMAC_SIGNATURE=false" to disable the verification feature.'
+        `[${adyenMerchantAccount}]: The "secretHmacKey" config variable is missing to be able to verify ` +
+          `notifications, please generate a secret HMAC key in Adyen Customer Area ` +
+          `or set "enableHmacSignature=false" to disable the verification feature.`
       )
   }
+}
+
```

Comment:
It would be interesting to see how many ctpProjects and adyenProjects are configured whenever I start up the notification server as a server log message

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/541/commits/9c0e71db6c64a41aab143e72c5ab26f66f0687ca

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/541/commits/d8037b2c2786c77d76a52cf25b629f60d87112e0